### PR TITLE
[Feature] Support preprocess in gson serialization

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

Some objects are not easy to be persisted directly with JSON, like google collect Range.
Through the preprocess function, these objects can be converted into easily persistent
objects before persistence, like String.
Then complete the deserialization through the postprocess function.
This makes gson serialization easier to use.

### CODE REVIEW

**Review by ABingHuang — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java [hunk 1]
```diff
@@ -426,8 +426,11 @@ public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
             TypeAdapter<T> delegate = gson.getDelegateAdapter(this, type);
 
             return new TypeAdapter<T>() {
-                public void write(JsonWriter out, T value) throws IOException {
-                    delegate.write(out, value);
+                public void write(JsonWriter out, T obj) throws IOException {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java:429]: Better to change the PostProcessTypeAdapterFactory to ProcessHookTypeAdapterFactory

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 2]
```diff
@@ -122,8 +123,6 @@ public enum OlapTableState {
 
     @SerializedName(value = "idToPartition")
     protected Map<Long, Partition> idToPartition = new HashMap<>();
-
-    @SerializedName(value = "nameToPartition")
     protected Map<String, Partition> nameToPartition = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:126]: If you remove the nameToPartition from the gson serializedName, you should make sure that this feature has not been released, or the nameToPartition will be empty after rollback fe.

File: fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java:429]: update

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Review by gengjun-git — APPROVED** [review]
LGTM

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|

**Pull request merged**